### DISCUSSION TITLE
Is there a DRY way to represent REST interfaces?

### DISCUSSION THREAD

**Discussion by trenta3** [vlang/v#11903] [Ideas]
Hi! I would like to know if there currently is a way to avoid boilerplate on standard REST CRUD interfaces by simply specifying models as structures.
Suppose I want to declare a API for Posts like
```v
struct Post {
    id int
    title string
    body string
}

$gen_create<App, Post>()
```
and I would like this call to generate code for the creation view like
```v
["/post"; post]
pub fun (mut app App) post_create () vweb.Result {
    // .... CODE HERE
}
```

Is this possible at the current time? Are there code creation facilities that could be used like this?

**Comment by spytheman** [comment]
> Is this possible at the current time?

Unfortunately not yet.

**Comment by igotfr** [comment]
related to: https://github.com/vlang/v/discussions/11732
